File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorConstraintNameAutoRefDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_constraint_name_auto_ref. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorConstraintNameAutoRefDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorConstraintNameAutoRefDbm _instance = new VendorConstraintNameAutoRefDbm();
    private VendorConstraintNameAutoRefDbm() {}
    public static VendorConstraintNameAutoRefDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoRefId(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoRefId(ctl(vl)), "constraintNameAutoRefId");
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoFooId(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoFooId(ctl(vl)), "constraintNameAutoFooId");
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoBarId(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoBarId(ctl(vl)), "constraintNameAutoBarId");
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoQuxId(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoQuxId(ctl(vl)), "constraintNameAutoQuxId");
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoCorgeId(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoCorgeId(ctl(vl)), "constraintNameAutoCorgeId");
        setupEpg(_epgMap, et -> ((VendorConstraintNameAutoRef)et).getConstraintNameAutoUnique(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setConstraintNameAutoUnique((String)vl), "constraintNameAutoUnique");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((VendorConstraintNameAutoRef)et).getVendorConstraintNameAutoBar(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setVendorConstraintNameAutoBar((OptionalEntity<VendorConstraintNameAutoBar>)vl), "vendorConstraintNameAutoBar");
        setupEfpg(_efpgMap, et -> ((VendorConstraintNameAutoRef)et).getVendorConstraintNameAutoFoo(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setVendorConstraintNameAutoFoo((OptionalEntity<VendorConstraintNameAutoFoo>)vl), "vendorConstraintNameAutoFoo");
        setupEfpg(_efpgMap, et -> ((VendorConstraintNameAutoRef)et).getVendorConstraintNameAutoQux(), (et, vl) -> ((VendorConstraintNameAutoRef)et).setVendorConstraintNameAutoQux((OptionalEntity<VendorConstraintNameAutoQux>)vl), "vendorConstraintNameAutoQux");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_constraint_name_auto_ref";
    protected final String _tableDispName = "vendor_constraint_name_auto_ref";
    protected final String _tablePropertyName = "vendorConstraintNameAutoRef";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_constraint_name_auto_ref", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnConstraintNameAutoRefId = cci("CONSTRAINT_NAME_AUTO_REF_ID", "CONSTRAINT_NAME_AUTO_REF_ID", null, null, Long.class, "constraintNameAutoRefId", null, true, false, true, "DECIMAL", 16, 0, null, false, null, null, null, null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnConstraintNameAutoFooId = cci("CONSTRAINT_NAME_AUTO_FOO_ID", "CONSTRAINT_NAME_AUTO_FOO_ID", null, null, Long.class, "constraintNameAutoFooId", null, false, false, true, "DECIMAL", 16, 0, null, false, null, null, "vendorConstraintNameAutoFoo", null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnConstraintNameAutoBarId = cci("CONSTRAINT_NAME_AUTO_BAR_ID", "CONSTRAINT_NAME_AUTO_BAR_ID", null, null, Long.class, "constraintNameAutoBarId", null, false, false, true, "DECIMAL", 16, 0, null, false, null, null, "vendorConstraintNameAutoBar", null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnConstraintNameAutoQuxId = cci("CONSTRAINT_NAME_AUTO_QUX_ID", "CONSTRAINT_NAME_AUTO_QUX_ID", null, null, Long.class, "constraintNameAutoQuxId", null, false, false, true, "DECIMAL", 16, 0, null, false, null, null, "vendorConstraintNameAutoQux", null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnConstraintNameAutoCorgeId = cci("CONSTRAINT_NAME_AUTO_CORGE_ID", "CONSTRAINT_NAME_AUTO_CORGE_ID", null, null, Long.class, "constraintNameAutoCorgeId", null, false, false, true, "DECIMAL", 16, 0, null, false, null, null, null, null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnConstraintNameAutoUnique = cci("CONSTRAINT_NAME_AUTO_UNIQUE", "CONSTRAINT_NAME_AUTO_UNIQUE", null, null, String.class, "constraintNameAutoUnique", null, false, false, true, "VARCHAR", 50, 0, null, false, null, null, null, null, null, false);
//"string"

    /**
     * CONSTRAINT_NAME_AUTO_REF_ID: {PK, NotNull, DECIMAL(16)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoRefId() { return _columnConstraintNameAutoRefId; }
    /**
     * CONSTRAINT_NAME_AUTO_FOO_ID: {IX, NotNull, DECIMAL(16), FK to vendor_constraint_name_auto_foo}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoFooId() { return _columnConstraintNameAutoFooId; }
    /**
     * CONSTRAINT_NAME_AUTO_BAR_ID: {IX, NotNull, DECIMAL(16), FK to vendor_constraint_name_auto_bar}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoBarId() { return _columnConstraintNameAutoBarId; }
    /**
     * CONSTRAINT_NAME_AUTO_QUX_ID: {IX, NotNull, DECIMAL(16), FK to vendor_constraint_name_auto_qux}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoQuxId() { return _columnConstraintNameAutoQuxId; }
    /**
     * CONSTRAINT_NAME_AUTO_CORGE_ID: {NotNull, DECIMAL(16)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoCorgeId() { return _columnConstraintNameAutoCorgeId; }
    /**
     * CONSTRAINT_NAME_AUTO_UNIQUE: {UQ, NotNull, VARCHAR(50)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnConstraintNameAutoUnique() { return _columnConstraintNameAutoUnique; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnConstraintNameAutoRefId());
        ls.add(columnConstraintNameAutoFooId());
        ls.add(columnConstraintNameAutoBarId());
        ls.add(columnConstraintNameAutoQuxId());
        ls.add(columnConstraintNameAutoCorgeId());
        ls.add(columnConstraintNameAutoUnique());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnConstraintNameAutoRefId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() { return hpcui(columnConstraintNameAutoUnique()); }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * vendor_constraint_name_auto_bar by my CONSTRAINT_NAME_AUTO_BAR_ID, named 'vendorConstraintNameAutoBar'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorConstraintNameAutoBar() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnConstraintNameAutoBarId(), VendorConstraintNameAutoBarDbm.getInstance().columnConstraintNameAutoBarId());
        return cfi("vendor_constraint_name_auto_ref_ibfk_2", "vendorConstraintNameAutoBar", this, VendorConstraintNameAutoBarDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorConstraintNameAutoRefList", false);
    }
    /**
     * vendor_constraint_name_auto_foo by my CONSTRAINT_NAME_AUTO_FOO_ID, named 'vendorConstraintNameAutoFoo'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorConstraintNameAutoFoo() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnConstraintNameAutoFooId(), VendorConstraintNameAutoFooDbm.getInstance().columnConstraintNameAutoFooId());
        return cfi("vendor_constraint_name_auto_ref_ibfk_1", "vendorConstraintNameAutoFoo", this, VendorConstraintNameAutoFooDbm.getInstance(), mp, 1, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorConstraintNameAutoRefList", false);
    }
    /**
     * vendor_constraint_name_auto_qux by my CONSTRAINT_NAME_AUTO_QUX_ID, named 'vendorConstraintNameAutoQux'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorConstraintNameAutoQux() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnConstraintNameAutoQuxId(), VendorConstraintNameAutoQuxDbm.getInstance().columnConstraintNameAutoQuxId());
        return cfi("vendor_constraint_name_auto_ref_ibfk_3", "vendorConstraintNameAutoQux", this, VendorConstraintNameAutoQuxDbm.getInstance(), mp, 2, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorConstraintNameAutoRefList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorConstraintNameAutoRef"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorConstraintNameAutoRefCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorConstraintNameAutoRefBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorConstraintNameAutoRef> getEntityType() { return VendorConstraintNameAutoRef.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorConstraintNameAutoRef newEntity() { return new VendorConstraintNameAutoRef(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorConstraintNameAutoRef)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorConstraintNameAutoRef)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/vendorConstraintNameAutoRefdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorConstraintNameAutoRefDbm_T struct {
	df.BaseDBMeta
	ColumnConstraintNameAutoRefId *df.ColumnInfo
	ColumnConstraintNameAutoFooId *df.ColumnInfo
	ColumnConstraintNameAutoBarId *df.ColumnInfo
	ColumnConstraintNameAutoQuxId *df.ColumnInfo
	ColumnConstraintNameAutoCorgeId *df.ColumnInfo
	ColumnConstraintNameAutoUnique *df.ColumnInfo
}

func (b *VendorConstraintNameAutoRefDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorConstraintNameAutoRefDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorConstraintNameAutoRefDbm *VendorConstraintNameAutoRefDbm_T

func Create_VendorConstraintNameAutoRefDbm() {
	VendorConstraintNameAutoRefDbm = new(VendorConstraintNameAutoRefDbm_T)
	VendorConstraintNameAutoRefDbm.TableDbName = "vendor_constraint_name_auto_ref"
	VendorConstraintNameAutoRefDbm.TableDispName = "vendor_constraint_name_auto_ref"
	VendorConstraintNameAutoRefDbm.TablePropertyName = "vendorConstraintNameAutoRef"
	VendorConstraintNameAutoRefDbm.TableSqlName = new(df.TableSqlName)
	VendorConstraintNameAutoRefDbm.TableSqlName.TableSqlName = "vendor_constraint_name_auto_ref"
	VendorConstraintNameAutoRefDbm.TableSqlName.CorrespondingDbName = VendorConstraintNameAutoRefDbm.TableDbName

	var vendorConstraintNameAutoRef df.DBMeta
	vendorConstraintNameAutoRef = VendorConstraintNameAutoRefDbm
	VendorConstraintNameAutoRefDbm.DBMeta=&vendorConstraintNameAutoRef
	constraintNameAutoRefIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_REF_ID
	constraintNameAutoRefIdSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_REF_ID"
	constraintNameAutoRefIdSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoRefId = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_REF_ID", constraintNameAutoRefIdSqlName, "", "", "Long.class", "constraintNameAutoRefId", "", true, false,true, "DECIMAL", 16, 0, "",false,"","", "","","",false,"df.Numeric")
	constraintNameAutoFooIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_FOO_ID
	constraintNameAutoFooIdSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_FOO_ID"
	constraintNameAutoFooIdSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoFooId = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_FOO_ID", constraintNameAutoFooIdSqlName, "", "", "Long.class", "constraintNameAutoFooId", "", false, false,true, "DECIMAL", 16, 0, "",false,"","", "vendorConstraintNameAutoFoo","","",false,"df.Numeric")
	constraintNameAutoBarIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_BAR_ID
	constraintNameAutoBarIdSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_BAR_ID"
	constraintNameAutoBarIdSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoBarId = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_BAR_ID", constraintNameAutoBarIdSqlName, "", "", "Long.class", "constraintNameAutoBarId", "", false, false,true, "DECIMAL", 16, 0, "",false,"","", "vendorConstraintNameAutoBar","","",false,"df.Numeric")
	constraintNameAutoQuxIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_QUX_ID
	constraintNameAutoQuxIdSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_QUX_ID"
	constraintNameAutoQuxIdSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoQuxId = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_QUX_ID", constraintNameAutoQuxIdSqlName, "", "", "Long.class", "constraintNameAutoQuxId", "", false, false,true, "DECIMAL", 16, 0, "",false,"","", "vendorConstraintNameAutoQux","","",false,"df.Numeric")
	constraintNameAutoCorgeIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_CORGE_ID
	constraintNameAutoCorgeIdSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_CORGE_ID"
	constraintNameAutoCorgeIdSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoCorgeId = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_CORGE_ID", constraintNameAutoCorgeIdSqlName, "", "", "Long.class", "constraintNameAutoCorgeId", "", false, false,true, "DECIMAL", 16, 0, "",false,"","", "","","",false,"df.Numeric")
	constraintNameAutoUniqueSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo CONSTRAINT_NAME_AUTO_UNIQUE
	constraintNameAutoUniqueSqlName.ColumnSqlName = "CONSTRAINT_NAME_AUTO_UNIQUE"
	constraintNameAutoUniqueSqlName.IrregularChar = false
	VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoUnique = df.CCI(&vendorConstraintNameAutoRef, "CONSTRAINT_NAME_AUTO_UNIQUE", constraintNameAutoUniqueSqlName, "", "", "String.class", "constraintNameAutoUnique", "", false, false,true, "VARCHAR", 50, 0, "",false,"","", "","","",false,"string")

	VendorConstraintNameAutoRefDbm.ColumnInfoList = new(df.List)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoRefId)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoFooId)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoBarId)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoQuxId)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoCorgeId)
	VendorConstraintNameAutoRefDbm.ColumnInfoList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoUnique)


	VendorConstraintNameAutoRefDbm.ColumnInfoMap=make(map[string]int)
	VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoRefId"]=0
		VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoFooId"]=1
		VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoBarId"]=2
		VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoQuxId"]=3
		VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoCorgeId"]=4
		VendorConstraintNameAutoRefDbm.ColumnInfoMap["constraintNameAutoUnique"]=5
	    VendorConstraintNameAutoRefDbm.PrimaryKey = true
    VendorConstraintNameAutoRefDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorConstraintNameAutoRef
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorConstraintNameAutoRefDbm.ColumnConstraintNameAutoRefId)

	VendorConstraintNameAutoRefDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorConstraintNameAutoRefDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorConstraintNameAutoRefMeta df.DBMeta = VendorConstraintNameAutoRefDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorConstraintNameAutoRef"] = &vendorConstraintNameAutoRefMeta
}
